src/tree_node.h

```diff
@@ -62,6 +69,21 @@ void TreeNode<V>::addChild(TreeNode<V>* node) {
   children.push_back(node);
 }
 
+template<typename V>
+void TreeNode<V>::setChild(TreeNode<V>* node, int index) {
+	if (index < children.size()) {
+		children[index] = node;
+	}
+	else {
+		// I calculate how many empty spaces to add to the list of children.

```

Comment:
Would like to have "// Calculates ... " instead of "// I calculate" as it describes what the code does, not what you are doing.